smnavigation/views.py

```diff
@@ -0,0 +1,65 @@
+from django.db import transaction
+from django.db.models.signals import post_save
+from django.dispatch import receiver
+from rest_framework import parsers, renderers, viewsets
+from rest_framework.authentication import SessionAuthentication
+from rest_framework.permissions import IsAuthenticated
+from rest_framework.response import Response
+from smasu.authentication import (
+    IsCreateView,
+    IsListView,
+    IsRetrieveView,
+    IsSuperUserOrStaff,
+    TokenAuthenticationInQuery,
+)
+from smasu.helpers import as_entity
+from smasu.models import Application
+from smasu.parsers import GeoJSONParser
+from smasu.renderers import GeoJSONRenderer
+from smevents.models import Event
+from smreports.models import SmartMovingEventType
+from smrouter.utils import Router
+
+from .models import NavigationRequest
+from .serializers import NavigationRequestSerializer
+
+
+class IsSmartMovingUser(IsAuthenticated):
+    def has_permission(self, request, view):
+        return super().has_permission(request, view) and request.user.smartmovingprofile is not None
+
+
+class NavigationRequestView(viewsets.ModelViewSet):
+    queryset = NavigationRequest.objects.all()
+    serializer_class = NavigationRequestSerializer
+    authentication_classes = (SessionAuthentication, TokenAuthenticationInQuery)
+    permission_classes = (IsSuperUserOrStaff | ((IsListView | IsCreateView | IsRetrieveView) & IsSmartMovingUser),)
+    parser_classes = (GeoJSONParser, parsers.FormParser)
+    renderer_classes = [GeoJSONRenderer, renderers.BrowsableAPIRenderer]
+
+    @receiver(post_save, sender=NavigationRequest)
+    def set(sender, instance, **kwargs):
+        router = Router()
+
+        with transaction.atomic():
+            if kwargs.get("created", False):
+                origin = instance.origin
+                destination = instance.destination
+                report_types_severe = instance.report_severe
```

Comment:
what's the purpose of these two `report_types_severe` and `report_types_light fields`?

Reply:
the configuration of each user , the list ids of the types of reports,  so we can calculate the additional cost of the reports. Eg. 

```user1``` has stair, bad sidewalk and no ramp as ```type_severe``` and the rest ```light```, so we give the ``cost X`` of each report severe and the ``cost Y`` of each report light that exist in the map. 
``` user2``` has uneven road, car in sidewalk and obstacle as ``light``, and the rest of the reports as ``severe``, so the reports light has the additional ```cost Y``` and the severes ```cost X ```

Its important this information, to calculate the the new cost for the ```navigation request```